### ISSUE TITLE
[ENH] cluster threshold for surface GLM

### ISSUE THREAD

**Issue by mszinte** [nilearn/nilearn#4326]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

As I use these days mostly (if not only) surface analysis of fMRI data, I use to the surface GLM analysis of nilearn, however I couldn't find the cluster_threshold method available for volume analysis. 
The idea of cluster analysis for surface would be to discard isolated vertices from the GLM output in order to generate thresholded maps with small cluster of vertices removed following a cluster_threshold argument.
Here is the volume based equivalent: 
https://github.com/nilearn/nilearn/blob/436cfb2196b69054d876e592a59c5a391b36f4c8/nilearn/glm/thresholding.py#L184
After early discussion I feel that for this improvement (@Remi-Gau @bthirion, @ulascombes) , we would need first:
* to associate each vertex to an area
* define connection component of each vertex
* set rules of connection of a vertices (maybe geodesic distances)

### Benefits to the change

This analysis is to my opinion an important missing step for the basic GLM surface analysis of nilearn, it will benefit to the community of users who moved toward fMRI surface analyses.

### Pseudocode for the new behavior, if applicable

```python
# insert your code below

```

**Comment by bthirion** [comment]
Well, you don't want to have to compute geodesic distances to obtain connected clusters. You simply need to use the topology of the mesh (corresponding to the face definition).
But I agree that this would be useful.

**Comment by Remi-Gau** [comment]
started working on this
- first step will be to identify clusters
- later on will work on actually doing stats on those clusters
- then will come getting report tables with just cluster label, size, stats. Never heard of a way to report surface cluster coordinates, but I am all ears.

**Comment by bthirion** [comment]
For coordinates, the best option to me is not to provide any.
If we really want, take the 3D coordinates of the vertices ?
For fsaverage pial/white, they're supposed to be in Talairach space.

**Comment by Remi-Gau** [comment]
> For coordinates, the best option to me is not to provide any.

yeah I would go for this for now, because most other options won't generalize well I think

**Comment by Remi-Gau** [comment]
- reopening given that #5764 is not done yet
